File: atcoder.jp/contests/abc178/tasks/abc178_b/main.py
from functools import reduce
from itertools import product
from operator import mul
from typing import Iterable


def prod(iterable: Iterable[int]) -> int:
    return reduce(mul, iterable)


A, B, C, D = map(int, input().split())

ans = max(map(prod, product([A, B], [C, D])))
print(ans)

File: atcoder.jp/contests/abc178/tasks/abc178_b/main.cpp
#include <bits/stdc++.h>
using namespace std;

using ll = long long;

int main() {
  ll A, B, C, D;
  cin >> A >> B >> C >> D;

  ll ans = max({A * C, A * D, B * C, B * D});
  cout << ans << endl;
}
